### PR TITLE
Add formula for gitup.

### PR DESCRIPTION
**Pull request by pr0d1r2**

### CODE REVIEW

File: Library/Formula/gitup.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Gitup < Formula
+  homepage "https://github.com/earwig/git-repo-updater"
+  url "https://github.com/earwig/git-repo-updater.git",
+    :revision => "51a482d7a25bed39ccc75780813c65bd666c1436"
```
Comment (DomT4) [inline Library/Formula/gitup.rb]: Please add the `:tag` as well as the `:revision`, Thanks! The addition of `tag` should make the `version` line redundant as well.

**Comment by DomT4** [comment]
I recommend glancing at the [jrnl](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/jrnl.rb) formula for how the Python elements should be styled. There's a couple things which need tweaking here.

**Comment by pr0d1r2** [comment]
@DomT4 now should be good :)

**Comment by DomT4** [comment]
The checksums need bumping to `sha256` but otherwise looks good.

```
==> audit problems
gitup:
 * Stable resource "smmap": SHA1 checksums are deprecated, please use SHA256
 * Stable resource "colorama": SHA1 checksums are deprecated, please use SHA256
 * Stable resource "gitdb": SHA1 checksums are deprecated, please use SHA256
 * Stable resource "GitPython": SHA1 checksums are deprecated, please use SHA256
```

File: Library/Formula/gitup.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    system "#{bin}/gitup", "-v"
```
Comment (DomT4) [inline Library/Formula/gitup.rb]: Can we improve this test? Upstream provide [some examples](https://github.com/earwig/git-repo-updater#usage), and you can use the system `git` to create temporary repos in the testpath, seen [here](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/git-stree.rb#L19-L35).

**Comment by pr0d1r2** [comment]
Great opportunity to learn write formulae proper way :D

File: Library/Formula/gitup.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      system "git", "reset", "--hard"
+      system "git", "checkout", "-b", "master"
+      system "git", "branch", "--set-upstream-to=origin/master", "master"
+    end
+
+    system "gitup", "first", "second"
+
+    first_head = `cd first ; git rev-parse HEAD`.split.first
+    if first_head == first_head_start
+      fail "FAILED: expect to change head from #{first_head_start}, but didn't"
```
Comment (DomT4) [inline Library/Formula/gitup.rb]: You can assert here and then it'll fail if the assertion doesn't match. We don't use `puts` in the `test do` blocks.

File: Library/Formula/gitup.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      system "git", "init"
+      system "git", "remote", "add", "origin", "https://github.com/pr0d1r2/homebrew-cask-games.git"
+      system "git", "fetch", "origin"
+      system "git", "checkout", second_head_start
+      system "git", "reset", "--hard"
+      system "git", "checkout", "-b", "master"
+      system "git", "branch", "--set-upstream-to=origin/master", "master"
+    end
+
+    system "gitup", "first", "second"
```
Comment (DomT4) [inline Library/Formula/gitup.rb]: `system bin/"gitup" `. We add the direct path to our executables in testing.

File: Library/Formula/gitup.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
+class Gitup < Formula
+  homepage "https://github.com/earwig/git-repo-updater"
+  url "https://github.com/earwig/git-repo-updater.git",
+    :revision => "51a482d7a25bed39ccc75780813c65bd666c1436",
+    :tag => "v0.2.3"
```
Comment (bfontaine) [inline Library/Formula/gitup.rb]: You could use only of them here. The `tag` points to the revision so you can use that directly.

Comment (DomT4) [inline Library/Formula/gitup.rb]: Afraid not. We mandate the `tag` and `revision` now - The reasoning being that the tag can be retagged silently which has bitten us before. The `revision` also acts as a checksum in that regard.

Comment (bfontaine) [inline Library/Formula/gitup.rb]: Thanks @DomT4, wasn’t aware of that.

File: Library/Formula/gitup.rb [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    end
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  include Test::Unit::Assertions
```
Comment (pr0d1r2) [inline Library/Formula/gitup.rb]: @DomT4 Add this to resolve testing issues with `assert_include` not present in formula. However tests seems not running at all ... issues with CI?

Comment (pr0d1r2) [inline Library/Formula/gitup.rb]: See http://bot.brew.sh/job/Homebrew%20Pull%20Requests/26702/version=mountain_lion/testReport/junit/brew-test-bot/mountain_lion/test_gitup/ for details.

Comment (DomT4) [inline Library/Formula/gitup.rb]: > Add this to resolve testing issues with assert_include not present in formula

You can just use `assert_match`, which will work without the extra include.

> for details.

The tests seem to be running there?

```
==> git init
Initialized empty Git repository in /private/tmp/gitup20150525-47435-t2uqw1/second/.git/
==> git remote add origin https://github.com/pr0d1r2/homebrew-cask-games.git
==> git fetch origin
From https://github.com/pr0d1r2/homebrew-cask-games
 * [new branch]      master     -> origin/master
==> git checkout f863d5ca9e39e524e8c222428e14625a5053ed2b
Note: checking out 'f863d5ca9e39e524e8c222428e14625a5053ed2b'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at f863d5c... Remove redundant not supported stanzas.
```

**Comment by pr0d1r2** [comment]
@DomT4 Requested changes are in place. However CI does not run (Looks like does not handle well `git push --force` after `git commit -a --amend`. Perhaps some kind of manual reset will be need on Jenkinks)

**Comment by pr0d1r2** [comment]
Pass `brew audit --strict gitup.rb`

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for new formulae is `gitup 0.2.3 (new formula)`.

**Comment by pr0d1r2** [comment]
@DomT4 & @mikemcquaid: Published some insights that may ease your work: http://pr0d1r2.tumblr.com/post/120171064468/achievement-unlocked-homebrew-contributor

**Pull request closed without merging**